dds/DCPS/DataWriterImpl.cpp

```diff
@@ -1520,7 +1520,8 @@ DataWriterImpl::register_instance_i(DDS::InstanceHandle_t& handle,
   if (ret != DDS::RETCODE_OK) {
     ACE_ERROR_RETURN((LM_ERROR,
                       ACE_TEXT("(%P|%t) ERROR: DataWriterImpl::register_instance_i: ")
-                      ACE_TEXT("register instance with container failed.\n")),
+                      ACE_TEXT("register instance with container failed, returned <%C>.\n"),
+                      retcode_to_string(ret)),
```

Comment:
```suggestion
                      retcode_to_string(ret).c_str()),
```

Reply:
Wouldn’t it be easier to let it return a const char*?

Reply:
They're all static strings so it seems like that would be good.  Fred?

Reply:
See https://github.com/DOCGroup/CIAO/blob/master/connectors/dds4ccm/impl/Utils.h for how we do this in DDS4CCM